packages/strapi-plugin-upload/admin/src/components/InputMedia/index.js

```diff
@@ -14,12 +17,21 @@ import Name from './Name';
 import Wrapper from './Wrapper';
 
 const InputMedia = ({ label, onChange, name, attribute, value, type }) => {
-  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [modal, setModal] = useState({
+    isOpen: false,
+    step: null,
+  });
   const [fileToDisplay, setFileToDisplay] = useState(0);
+  const hasNoValue = !!value && Array.isArray(value) && value.length === 0;
+  const currentFile = attribute.multiple ? value[fileToDisplay] : value;
+  const fileURL = get(currentFile, ['url'], null);
+  const prefixedFileURL = fileURL ? prefixFileUrlWithBackendUrl(fileURL) : null;
+  const displaySlidePagination =
+    attribute.multiple && value.length > 1 ? ` (${fileToDisplay + 1}/${value.length})` : '';
+
   const handleClickToggleModal = () => {
-    setIsModalOpen(prev => !prev);
+    setModal(prev => ({ isOpen: !prev.isOpen }));
```

Comment:
You are losing some state here:

```
 setModal(prev => ({ ...prev, isOpen: !prev.isOpen }));
```

Reply:
It is useless here. The modal step is the Browse list by default.

Reply:
Yes but you loose the step when this is called